### PR TITLE
Add a new multi-line label component that supports truncating its contents

### PR DESCRIPTION
**Pull request by sophistifunk**
**Description**

* Add a new TruncatingLabel component that cuts down its text to fit and adds ellipses if too long.
* Update PipelineGraph to use TruncatingLabel in place of plain `<div>`s for node labels.

- See [JENKINS-38753](https://issues.jenkins-ci.org/browse/JENKINS-38753).

**Submitter checklist**
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by sophistifunk** [comment]
screen shot 2017-02-15 at 5 55 28 pm

**Comment by michaelneale** [comment]
looks good in story book - will see if others can review code cc @cliffmeyers @kzantow @scherler

File: src/js/components/TruncatingLabel.jsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    //  Render
+    //--------------------------------------
+
+    render() {
+        const {
+            style = {},
+            className = ''
+        } = this.props;
+
+        const mergedStyle = {
```
Comment (kzantow) [inline src/js/components/TruncatingLabel.jsx]: not sure why mixed `className` and `style` here, maybe just get rid of `style`?

Comment (cliffmeyers) [inline src/js/components/TruncatingLabel.jsx]: I think it's good to support both, just like any HTML element would.

Comment (sophistifunk) [inline src/js/components/TruncatingLabel.jsx]: All components should reflect incoming className and style props onto the divs they generate.

File: src/js/components/TruncatingLabel.jsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+            let keepMeasuring;
+
+            if (this.loopCount >= MAXLOOPS) {
+                // This really shouldn't happen!
+                console.error('TruncatingLabel - TOO MANY LOOPS');
+                keepMeasuring = false;
+            } else if (lastLength <= MINLENGTH) {
+                keepMeasuring = false;
+            } else if (Math.abs(this.shortestBad - this.longestGood) < 2) {
```
Comment (cliffmeyers) [inline src/js/components/TruncatingLabel.jsx]: Do I interpret this line correctly as "there are no more suitable lengths to try, so stop measuring"? A comment might help explain that - it took me a moment.

Comment (sophistifunk) [inline src/js/components/TruncatingLabel.jsx]: Good idea

**Pull request merged**